android/KMEA/app/src/main/java/com/tavultesoft/kmea/KeyboardPickerActivity.java

```diff
@@ -252,41 +237,21 @@ public void onBackPressed() {
     finish();
   }
 
-  private static int getCurrentKeyboardIndex() {
+  /**
+   * Get the current keyboard index. Only to use for KeyboardPickerActivity or unit test.
+   * @return int of the matching keyboard index. If current index not found, return 0
+   */
+  public static int getCurrentKeyboardIndex() {
     int pos = 0;
+    pos = KeyboardController.getInstance().getKeyboardIndex(KMKeyboard.currentKeyboard());
 
-    if (keyboardsList != null) {
-      int length = keyboardsList.size();
-      for (int i = 0; i < length; i++) {
-        HashMap<String, String> kbInfo = keyboardsList.get(i);
-        String langId = kbInfo.get(KMManager.KMKey_LanguageID);
-        String kbId = kbInfo.get(KMManager.KMKey_KeyboardID);
-        String kbKey = String.format("%s_%s", langId, kbId);
-        if (kbKey.equals(KMKeyboard.currentKeyboard())) {
-          pos = i;
-          break;
-        }
-      }
-    }
-
-    return pos;
-  }
-
-  // Previously, keyboard from package = custom keyboard
-  // Now, keyboard from package has a package ID != "cloud"
-  protected static boolean hasKeyboardFromPackage() {
-    for(HashMap<String, String> kbInfo: keyboardsList) {
-      String pkgID = MapCompat.getOrDefault(kbInfo, KMManager.KMKey_PackageID, KMManager.KMDefault_UndefinedPackageID);
-      if (!pkgID.equalsIgnoreCase(KMManager.KMDefault_UndefinedPackageID)) {
-        return true;
-      }
-    }
-
-    return false;
+    // If index not found, return first index
+    return (pos != KeyboardController.INDEX_NOT_FOUND) ? pos : 0;
```

Comment:
What is the index of the first item in the list?

Reply:
The first item has index 0

Reply:
Okay, the comments need to clarify that it always returns a valid index. Are there cases where it might return 0 but there are actually no keyboards in the list?